### PR TITLE
Fix daemonset flapping when using a host port

### LINKED ISSUE #22309: Daemonset keeps getting killed if hostNetwork: true
I just built from HEAD.

The DS works fine with pod networking. As soon as I run it with hostNetwork: true, though, the container gets created, then a second or two later, kubelet says "Killing unwanted pod X".

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #22309

### CODE REVIEW

**Comment by mikedanese** [comment]
The diff breaks up the unit test weirdly be before the change it fails. We observe a 0 creates and 1 delete when we should observer 0 creates and 0 deletes

**Comment by ncdc** [comment]
cc @kubernetes/rh-cluster-infra

File: pkg/controller/daemon/controller.go [hunk 1]
```diff
@@ -601,6 +601,11 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *api.Node, ds *exte
 		if pod.Spec.NodeName != node.Name {
 			continue
 		}
+		// ignore pods that belong to the daemonset when taking into account wheter
```
Comment (aveshagarwal) [inline pkg/controller/daemon/controller.go]: typo: wheter

**Comment by davidopp** [comment]
LGTM

OK I read the code more carefully, now I understand.

**Pull request merged**